x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityLicensingAndFeatureUsageRestIT.java

```diff
@@ -160,14 +160,7 @@ public void testCrossClusterAccessFeatureTrackingAndLicensing() throws Exception
             );
 
             // Check that CCS fails because we cannot establish connection due to the license check.
-            if (useProxyMode) {
-                // TODO: We should improve error handling so we get actual cause instead just NoSeedNodeLeftException.
-                var exception = expectThrows(ResponseException.class, () -> performRequestWithRemoteSearchUser(searchRequest));
-                assertThat(exception.getResponse().getStatusLine().getStatusCode(), equalTo(500));
```

Comment:
For proxy mode, we still get a `500` here, which is why [testCrossClusterAccessFeatureTrackingAndLicensing is failing](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+part-3/13310/console). However, we only get a 500 when the FC fails the license check (since the handshake fails). When the QC fails the license check we still get a 403. 

I'm looking through the code still to figure out if this is the behavior we actually want (i.e., the inconsistency between proxy and sniff mode on FC-license check failure).